src/commands/moderation/move.js

```diff
@@ -8,42 +8,45 @@ module.exports = class Move extends Command {
       category: 'moderation',
       requirements: { guildOnly: true, botPermissions: ['MANAGE_MESSAGES'], permissions: ['MANAGE_MESSAGES'] },
       parameters: [{
-        type: 'messageLink', sameGuildOnly: true, forceExists: true, missingError: 'commands:move.missingMessageLink'
+        type: 'messageLink',
+        sameGuildOnly: true,
+        forceExists: true,
+        linkChannelUserPermission: ['VIEW_MESSAGES'],
+        linkChannelBotPermission: ['MANAGE_MESSAGES'],
+        missingError: 'commands:move.missingMessageLink'
       }, {
-        type: 'channel', onlySameGuild: true, acceptText: true, missingError: 'commands:move.missingChannelMention'
+        type: 'channel',
+        onlySameGuild: true,
+        acceptText: true,
+        channelUserPermission: ['SEND_MESSAGES'],
+        channelBotPermission: ['SEND_MESSAGES', 'EMBED_LINKS'],
+        missingError: 'commands:move.missingChannelMention'
       }]
     }, client)
   }
 
   async run ({ channel, guild, author, t, message }, link, destinationChannel) {
-    if (!destinationChannel.permissionsFor(this.client.user.id).has('SEND_MESSAGES')) throw new CommandError(t('commands:move.cantSendInChannel'))
+    if (link.channel.nsfw && !destinationChannel.nsfw) throw new CommandError(t('commands:move.channelsHaveDifferentType'))
 
-    try {
-      if (link.channel.nsfw && !destinationChannel.nsfw) throw new CommandError(t('commands:move.channelsHaveDifferentType'))
-
-      if (!link.channel.permissionsFor(this.client.user.id).has('MANAGE_MESSAGES')) throw new CommandError(t('commands:move.cantDeleteMessage'))
-
-      if (!destinationChannel.permissionsFor(author.id).has('SEND_MESSAGES')) throw new CommandError(t('commands:move.userCantSend'))
-
-      const MessageObj = {}
-      let content = link.content
+    const MessageObj = {}
+    let content = link.content
 
-      if (link.attachments.size >= 1) {
-        MessageObj.files = [link.attachments.first().url]
-      }
+    if (link.attachments.size >= 1) {
+      MessageObj.files = [link.attachments.first().url]
```

Comment:
hm, what if there are more than one attachment? i don't remember if bots are allowed to send more than one attachment tho

Reply:
they are

Reply:
yes, messages ik, i'm just not sure about bots, but it would be better to resend all the attachments like:

```js
MessageObj.files = link.attachments.array()
```

Reply:
> they are
> 
> actually, I don't think that there's a limit to the amount of attachments you can send in a message (probably file size only)

There's one for users that I'm aware.

Selecting and trying to send more than 20 images in the gallery prompts an error.

Reply:
> There's one for users that I'm aware.
> 
> Selecting and trying to send more than 20 images in the gallery prompts an error.

yeah, but it's better to just send the attachments as an array, even if it
is just one